pkg/vaulturl/helpers.go

```diff
@@ -0,0 +1,54 @@
+package vaulturl
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+)
+
+var vaultURIRegex = regexp.MustCompile(`vault:[-_\w\/:]*`)
```

Comment:
I would be explicit a use a URI scheme per client (e.g. vault:..., local:..., k8s:...) instead of hiding this. In this way, the user will be aware of the place where the secrets are stored.

Reply:
BTW one of the motivations of this was so that we don't have to search / replace all the values.yaml in a GitOps install if someone decides to switch from local file system secrets to vault - i.e. to use the same URI scheme in values.yaml

Reply:
I'm hoping we can have a canonical git repo for installing Jenkins X with prow + tekton with + without vault without having to do a global replace of `vault:` <-> `local:`

Reply:
I would use `secret` instead. It will be difficult to use the same path for both vault and local, because vault is using a relative path to the root of the secrets backend which normally is mounted with a fix name `secret` for KV secret engine. It will be something like `secret/<path-in-vault>:secret-name` but the `secret` prefix is omitted from the URI, which is not totally correct if we want to support multiple vault secrets backends. Another backend will be mounted on a different prefix so the user won't be able to use that backend.

I think the user will be forced to store the secrets file to the same relative path which is similar with the relative path in vault.

Given this example:
https://github.com/jenkins-x/environment-tekton-mole-dev/blob/783a8f6178c050c0b96c61016d1581bd0bc7a3a0/env/values.yaml#L98

the user will have to store the secrets on a local path like this:
`<root-folder>/gitOps/jenkins-x/environment-tekton-mole-dev/jenkins-x-demo-sa:data`